lib/Monocle_Alternate.php

```diff
@@ -235,10 +236,16 @@ public function query_gyms($conds, $params)
         r.move_1 AS raid_pokemon_move_1,
         r.move_2 AS raid_pokemon_move_2
         FROM forts f
-        LEFT JOIN fort_sightings fs ON fs.fort_id = f.id
-        LEFT JOIN raids r ON r.fort_id = f.id
+        LEFT JOIN fort_sightings fs ON (fs.fort_id = f.id AND :fort_condition)
+        LEFT JOIN raids r ON (r.fort_id = f.id AND r.time_end >= :time)
         WHERE :conditions";
 
+        if ($alternateKeepGymHistory) {
+            $query = str_replace(":fort_condition", "fs.last_modified = (SELECT MAX(last_modified) FROM fort_sightings fs2 WHERE fs2.fort_id=f.id)", $query);
+        } else {
+            $query = str_replace(":fort_condition", "1=1", $query);
```

Comment:
I don't like the `1=1` check.
Maybe include the `AND` inside `:fort_condition` and replace it with an empty string if `$alternateKeepGymHistory` is false?